evolver/device.py

```diff
@@ -92,3 +92,12 @@ def loop_once(self):
             self.evaluate_controllers()
         if self.enable_commit:
             self.commit_proposals()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.config_model.save(app_settings.CONFIG_FILE)
+
```

Comment:
We could also add here...
```suggestion
  self.enable_control = False
  self.enable_commit = False
```